### PR TITLE
Add support for multiple entry points in the ProjectGraph

### PR DESCRIPTION
**Pull request by dfederm**
Closes: #3795

### CODE REVIEW

File: src/Build.UnitTests/Graph/ProjectGraph_Tests.cs [hunk 1]
```diff
@@ -302,6 +301,27 @@ public void ConstructWithInvalidProperties()
             }
         }
 
+        [Fact]
+        public void ConstructWithMultipleEntryPoints()
```
Comment (rainersigwald) [inline src/Build.UnitTests/Graph/ProjectGraph_Tests.cs]: I'd also like to see same-project-different-properties as a test case.

File: src/Build/Graph/ProjectGraphEntryPoint.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        {
+            foreach (string entryProjectFile in entryProjectFiles)
+            {
+                yield return new ProjectGraphEntryPoint(entryProjectFile, globalProperties);
+            }
+        }
+
+        internal IEnumerable<ProjectGraphEntryPoint> ToEnumerable()
+        {
+            yield return this;
```
Comment (rainersigwald) [inline src/Build/Graph/ProjectGraphEntryPoint.cs]: This is a fun trick. Is it standard? I feel like I'd prefer the name `AsEnumerable` to emphasize that it's still a single item.

File: ref/Microsoft.Build/net/Microsoft.Build.cs [hunk 3]
```diff
@@ -1409,13 +1409,30 @@ namespace Microsoft.Build.Graph
...
+        public ProjectGraph(System.Collections.Generic.IEnumerable<Microsoft.Build.Graph.ProjectGraphEntryPoint> entryPoints) { }
+        public ProjectGraph(System.Collections.Generic.IEnumerable<Microsoft.Build.Graph.ProjectGraphEntryPoint> entryPoints, Microsoft.Build.Evaluation.ProjectCollection projectCollection) { }
+        public ProjectGraph(System.Collections.Generic.IEnumerable<string> entryProjectFiles) { }
+        public ProjectGraph(System.Collections.Generic.IEnumerable<string> entryProjectFiles, Microsoft.Build.Evaluation.ProjectCollection projectCollection) { }
+        public ProjectGraph(System.Collections.Generic.IEnumerable<string> entryProjectFiles, System.Collections.Generic.IDictionary<string, string> globalProperties) { }
+        public ProjectGraph(System.Collections.Generic.IEnumerable<string> entryProjectFiles, System.Collections.Generic.IDictionary<string, string> globalProperties, Microsoft.Build.Evaluation.ProjectCollection projectCollection) { }
         public ProjectGraph(string entryProjectFile) { }
+        public ProjectGraph(string entryProjectFile, Microsoft.Build.Evaluation.ProjectCollection projectCollection) { }
```
Comment (rainersigwald) [inline ref/Microsoft.Build/net/Microsoft.Build.cs]: Do we need this one, since you could always pass `null` as the properties dict?

Comment (dfederm) [reply ref/Microsoft.Build/net/Microsoft.Build.cs]: You could pass `null`, but really we don't "need" any of these ctors except the single one that actually does work. I figure all these are just for convenience.

It also is less clear on the calling side (unless you use named params) to just have:
```
new ProjectGraph(file, null, projectCollection);
```

"What's 'null' here?" I can see code reviewers asking.

File: src/Build.UnitTests/Graph/ProjectGraph_Tests.cs (hunk 1 above)
Comment (dfederm) [reply src/Build.UnitTests/Graph/ProjectGraph_Tests.cs]: Oooh, and a test with `GlobalPropertiesToRemove` causing convergence, ie the "agnostic project" scenario.

File: src/Build/Graph/ProjectGraphEntryPoint.cs (hunk 2 above)
Comment (dfederm) [reply src/Build/Graph/ProjectGraphEntryPoint.cs]: No idea if it's standard :). I assume it's better than wrapping it in an array, but I don't actually know that.

**Pull request merged**